status-im/status-react
# feat: enable biometrics for standard-auth

fixes: https://github.com/status-im/status-mobile/issues/17445

This pr adds biometrics for standard authentication -

https://www.figma.com/file/V6nlpAWIf2e1XU8RJ9yQPe/Syncing-for-Mobile?type=design&node-id=1074-118387&mode=design&t=XdGbtSuHsABYU9C7-0
Wallet:
https://www.figma.com/file/HKncH4wnDwZDAhc4AryK8Y/Wallet-for-Mobile?type=design&node-id=1741-374592&mode=design&t=DEQa5r3JJAPDGMKj-0

Currently this is used in the wallet to create an account and the syncing page to get a Sync code.

**To test:**
- With Biometrics enabled-
-- Create an account with biometrics enabled ->
Go to  new wallet, create a new account

-- Create a watch-only account  with biometrics enabled 
Go to  new wallet, create add an address to watch.
Authorization is not needed but this stil should be tested for this pr.

-- Syncing, generate a new sync code. 
 Go to generate sync code page and generate

I also put in the error handling for when the biometrics fails multiple times and fallsback to password. 

The same tests should also be run without biometrics enabled.

There are so many tests paths so let me know if I missed any 🙏
